File: services/business-services/menu-v3/infrastructure/repository/menu_option_repository.go
package repository

import (
	"menu-v3/infrastructure/repository/entity"
)

type MenuOptionRepository interface {
	FindAll() ([]entity.MenuOption, error)
	FindById(id uint) (*entity.MenuOption, error)
	Save(menuOption *entity.MenuOption) error
	Delete(id uint) error
}

File: application/backend/business/menu-v1/src/main/java/com/demo/bbq/business/menu/domain/repository/menuoption/MenuOptionRepository.java
package com.demo.bbq.business.menu.domain.repository.menuoption;

import com.demo.bbq.business.menu.domain.repository.menuoption.entity.MenuOptionEntity;
import java.util.List;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MenuOptionRepository extends CrudRepository<MenuOptionEntity, Long> {

  List<MenuOptionEntity> findAll();

  Optional<MenuOptionEntity> findById(Long id);

  Optional<MenuOptionEntity> findByProductCode(String productCode);

  List<MenuOptionEntity> findByCategory(String category);

  MenuOptionEntity save(MenuOptionEntity menuOptionEntity);

  void deleteByProductCode(String productCode);
}
